File: src/app/api/skills.py
from fastapi import APIRouter

from app.api.schemas import SkillPackageView, SkillsResponse
from app.services.skill_service import SkillService

router = APIRouter()
_skill_service = SkillService()


@router.get("/api/skills")
def list_skills() -> SkillsResponse:
    return SkillsResponse(
        skills=[
            SkillPackageView(
                name=skill.name,
                description=skill.description,
                path=skill.path,
                valid=skill.valid,
                error=skill.error,
                updated_at=skill.updated_at,
                body_size=skill.body_size,
            )
            for skill in _skill_service.list_skills()
        ]
    )

File: web/src/api/skills.ts
import { requestJson } from './client'
import type { SkillPackageDto, SkillsResponseDto } from '../types/api'
import type { SkillPackage } from '../types/ops'

export function mapSkillPackage(dto: SkillPackageDto): SkillPackage {
  return {
    name: dto.name,
    description: dto.description,
    path: dto.path,
    valid: dto.valid,
    error: dto.error,
    updatedAt: dto.updated_at,
    bodySize: dto.body_size,
  }
}

export async function getSkills(): Promise<SkillPackage[]> {
  const response = await requestJson<SkillsResponseDto>('/api/skills')
  return response.skills.map(mapSkillPackage)
}
